### DISCUSSION TITLE
How to configure GitVersion for automatic main version increase after "unversioned" hotfix merge?

### DISCUSSION THREAD

**Discussion by user1584** [GitTools/GitVersion#3815] [Q&A] [answered]
I try to configure GitVersion to follow the [GitFlow](https://datasift.github.io/gitflow/IntroducingGitFlow.html) strategy with as little human versioning as possible. According to the [GitVersion documentation](https://gitversion.net/docs/learn/branching-strategies/gitflow/examples#hotfix-branches), the hotfix branches should be named with the desired version for the main branch after merge (e.g. if main is on "1.2.0", the hotfix branch should be named "hotfix/12.1"). However, I was wondering if it is somehow possible to configure GitVersion such that all hotfix merges increase the patch part of the main branch version.   
What I am trying to achieve is that a name such as "hotfix/put-out-the-fire" will increase the main version from "1.2.0" to "1.2.1".  
Is this possible with 5.x or 6.x?

**Comment by HHobeck** [comment, marked as answer]
At this moment it is only supported in Mainline mode (in 5.x and 6-beta.3) with dedicated merge commit messages (in azure devops the source branch is missing and not working out of the box). In the 6-beta.4 version the Mainline mode will be properly replaced with a TrunkBased version strategy.

https://github.com/GitTools/GitVersion/blob/0ff62cc2aee284b32e85833aa69cc6081876a9cf/src/GitVersion.Core.Tests/IntegrationTests/MainlineDevelopmentMode.cs#L415

Please see also:
- https://github.com/GitTools/GitVersion/discussions/3800
- https://github.com/GitTools/GitVersion/discussions/3601
- https://github.com/GitTools/GitVersion/issues/3689

Happy Branching!

**Reply by user1584** [reply]
OK, would this be a reasonable improvement request?

**Reply by HHobeck** [reply]
Yes of course. We have already an improvement request. Please see #3689.

**Reply by user1584** [reply]
As far as I understand #3689, it's focus is on correct versioning of the development branch, whereas I am more concerned with the main branch. Is the target branch irrelevant? If yes, this improvement would be exacatly what I need.

**Reply by HHobeck** [reply]
Everything in GitVersion related to the behavior of how the versions are incremented are configurable. The whole feature is not existing and needs to be implement first (If someone will be found doing this). Afterwards we can think about how to use it in which workflows and on which branches. Any time you can tailor the workflow like you wish. Why not just document your thoughts in #3689 that you need this on main branch as well? I think a new issue is not needed.

Edit: Please provide integration tests for your scenario if you not mind.

**Reply by user1584** [reply]
OK, thanks! I have no experience with C# - therefore I am not sure how to write a test and how to test the test. Can you point me to some examples (specific to GitVersion)?

**Reply by HHobeck** [reply]
No problem. You will find the integration tests in the following folder: https://github.com/GitTools/GitVersion/tree/main/src/GitVersion.Core.Tests/IntegrationTests
